File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/builtin/UaDataTypes.java
package org.opcfoundation.webserver.addressspace.nodes.builtin;

import org.eclipse.milo.opcua.stack.core.NodeIds;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.webserver.addressspace.nodes.UaDataType;

public class UaDataTypes {
    public final static UaDataType BaseDataType = new UaDataType(NodeIds.BaseDataType,"BaseDataType", new LocalizedText("BaseDataType"), true);
    public final static UaDataType Boolean = new UaDataType(NodeIds.Boolean,"Boolean", new LocalizedText("Boolean"), false);

    public final static UaDataType Number = new UaDataType(NodeIds.Number,"Number", new LocalizedText("Number"), true);
    public final static UaDataType Integer = new UaDataType(NodeIds.Integer,"Integer", new LocalizedText("Integer"), true);
    public final static UaDataType SByte = new UaDataType(NodeIds.SByte,"SByte", new LocalizedText("SByte"), false);
    public final static UaDataType Int16 = new UaDataType(NodeIds.Int16,"Int16", new LocalizedText("Int16"), false);
    public final static UaDataType Int32 = new UaDataType(NodeIds.Int32,"Int32", new LocalizedText("Int32"), false);
    public final static UaDataType Int64 = new UaDataType(NodeIds.Int64,"Int64", new LocalizedText("Int64"), false);

    public final static UaDataType UInteger = new UaDataType(NodeIds.UInteger,"UInteger", new LocalizedText("UInteger"), true);
    public final static UaDataType Byte = new UaDataType(NodeIds.Byte,"Byte", new LocalizedText("Byte"), false);
    public final static UaDataType UInt16 = new UaDataType(NodeIds.UInt16,"UInt16", new LocalizedText("UInt16"), false);
    public final static UaDataType UInt32 = new UaDataType(NodeIds.UInt32,"UInt32", new LocalizedText("UInt32"), false);
    public final static UaDataType UInt64 = new UaDataType(NodeIds.UInt64,"UInt64", new LocalizedText("UInt64"), false);

    public final static UaDataType Float = new UaDataType(NodeIds.Float,"Float", new LocalizedText("Float"), false);
    public final static UaDataType Double = new UaDataType(NodeIds.Double,"Double", new LocalizedText("Double"), false);
    public final static UaDataType Duration = new UaDataType(NodeIds.Duration,"Duration", new LocalizedText("Duration"), false);
    public final static UaDataType Decimal = new UaDataType(NodeIds.Decimal,"Decimal", new LocalizedText("Decimal"), false);

    public final static UaDataType String = new UaDataType(NodeIds.String,"String", new LocalizedText("String"), false);
    public final static UaDataType DateString = new UaDataType(NodeIds.DateString,"DateString", new LocalizedText("DateString"), false);
    public final static UaDataType NormalizedString = new UaDataType(NodeIds.NormalizedString,"NormalizedString", new LocalizedText("NormalizedString"), false);
    public final static UaDataType NumericRange = new UaDataType(NodeIds.NumericRange,"NumericRange", new LocalizedText("NumericRange"), false);
    public final static UaDataType UriString = new UaDataType(NodeIds.UriString,"UriString", new LocalizedText("UriString"), false);

    public final static UaDataType ByteString = new UaDataType(NodeIds.ByteString,"ByteString", new LocalizedText("ByteString"), false);
    public final static UaDataType Image = new UaDataType(NodeIds.Image,"Image", new LocalizedText("Image"), true);
    public final static UaDataType ImageBMP = new UaDataType(NodeIds.ImageBMP,"ImageBMP", new LocalizedText("ImageBMP"), false);
    public final static UaDataType ImageGIF = new UaDataType(NodeIds.ImageGIF,"ImageGIF", new LocalizedText("ImageGIF"), false);
    public final static UaDataType ImageJPG = new UaDataType(NodeIds.ImageJPG,"ImageJPG", new LocalizedText("ImageJPG"), false);
    public final static UaDataType ImagePNG = new UaDataType(NodeIds.ImagePNG,"ImagePNG", new LocalizedText("ImagePNG"), false);
    public final static UaDataType AudioDataType = new UaDataType(NodeIds.AudioDataType,"AudioDataType", new LocalizedText("AudioDataType"), false);

    public final static UaDataType Enumeration = new UaDataType(NodeIds.Enumeration,"Enumeration", new LocalizedText("Enumeration"), true);

    public final static UaDataType DateTime = new UaDataType(NodeIds.DateTime,"DateTime", new LocalizedText("DateTime"), false);
    public final static UaDataType UtcTime = new UaDataType(NodeIds.UtcTime,"UtcTime", new LocalizedText("UtcTime"), false);

    public final static UaDataType NodeId = new UaDataType(NodeIds.NodeId,"NodeId", new LocalizedText("NodeId"), false);
    public final static UaDataType ExpandedNodeId = new UaDataType(NodeIds.ExpandedNodeId,"ExpandedNodeId", new LocalizedText("ExpandedNodeId"), false);
    public final static UaDataType Guid = new UaDataType(NodeIds.Guid,"Guid", new LocalizedText("Guid"), false);
    public final static UaDataType LocalizedText = new UaDataType(NodeIds.LocalizedText,"LocalizedText", new LocalizedText("LocalizedText"), false);
    public final static UaDataType QualifiedName = new UaDataType(NodeIds.QualifiedName,"QualifiedName", new LocalizedText("QualifiedName"), false);
    public final static UaDataType StatusCode = new UaDataType(NodeIds.StatusCode,"StatusCode", new LocalizedText("StatusCode"), false);

    public final static UaDataType Structure = new UaDataType(NodeIds.Structure,"Structure", new LocalizedText("Structure"), true);
    public final static UaDataType Range = new UaDataType(NodeIds.Range,"Range", new LocalizedText("Range"), false);
    public final static UaDataType EUInformation = new UaDataType(NodeIds.EUInformation,"EUInformation", new LocalizedText("EUInformation"), false);
    public final static UaDataType EnumValueType = new UaDataType(NodeIds.EnumValueType,"EnumValueType", new LocalizedText("EnumValueType"), false);
}

File: opcua-webservice-node/source/src/server/addressspace/nodes/builtin/UaDataTypes.ts
import { DataTypeIds, UaLocalizedText, UaNodeId } from "opcua-webapi-ts";
import { UaDataType } from "../UaDataType";

export class UaDataTypes {
    static readonly BaseDataType = new UaDataType(UaNodeId.from(DataTypeIds.BaseDataType), "BaseDataType", new UaLocalizedText("BaseDataType"), true);
    static readonly Boolean = new UaDataType(UaNodeId.from(DataTypeIds.Boolean), "Boolean", new UaLocalizedText("Boolean"), false);

    static readonly Number = new UaDataType(UaNodeId.from(DataTypeIds.Number), "Number", new UaLocalizedText("Number"), true);
    static readonly Integer = new UaDataType(UaNodeId.from(DataTypeIds.Integer), "Integer", new UaLocalizedText("Integer"), true);
    static readonly SByte = new UaDataType(UaNodeId.from(DataTypeIds.SByte), "SByte", new UaLocalizedText("SByte"), false);
    static readonly Int16 = new UaDataType(UaNodeId.from(DataTypeIds.Int16), "Int16", new UaLocalizedText("Int16"), false);
    static readonly Int32 = new UaDataType(UaNodeId.from(DataTypeIds.Int32), "Int32", new UaLocalizedText("Int32"), false);
    static readonly Int64 = new UaDataType(UaNodeId.from(DataTypeIds.Int64), "Int64", new UaLocalizedText("Int64"), false);

    static readonly UInteger = new UaDataType(UaNodeId.from(DataTypeIds.UInteger), "UInteger", new UaLocalizedText("UInteger"), true);
    static readonly Byte = new UaDataType(UaNodeId.from(DataTypeIds.Byte), "Byte", new UaLocalizedText("Byte"), false);
    static readonly UInt16 = new UaDataType(UaNodeId.from(DataTypeIds.UInt16), "UInt16", new UaLocalizedText("UInt16"), false);
    static readonly UInt32 = new UaDataType(UaNodeId.from(DataTypeIds.UInt32), "UInt32", new UaLocalizedText("UInt32"), false);
    static readonly UInt64 = new UaDataType(UaNodeId.from(DataTypeIds.UInt64), "UInt64", new UaLocalizedText("UInt64"), false);

    static readonly Float = new UaDataType(UaNodeId.from(DataTypeIds.Float), "Float", new UaLocalizedText("Float"), false);
    static readonly Double = new UaDataType(UaNodeId.from(DataTypeIds.Double), "Double", new UaLocalizedText("Double"), false);
    static readonly Duration = new UaDataType(UaNodeId.from(DataTypeIds.Duration), "Duration", new UaLocalizedText("Duration"), false);
    static readonly Decimal = new UaDataType(UaNodeId.from(DataTypeIds.Decimal), "Decimal", new UaLocalizedText("Decimal"), false);

    static readonly String = new UaDataType(UaNodeId.from(DataTypeIds.String), "String", new UaLocalizedText("String"), false);
    static readonly DateString = new UaDataType(UaNodeId.from(DataTypeIds.DateString), "DateString", new UaLocalizedText("DateString"), false);
    static readonly NormalizedString = new UaDataType(UaNodeId.from(DataTypeIds.NormalizedString), "NormalizedString", new UaLocalizedText("NormalizedString"), false);
    static readonly NumericRange = new UaDataType(UaNodeId.from(DataTypeIds.NumericRange), "NumericRange", new UaLocalizedText("NumericRange"), false);
    static readonly UriString = new UaDataType(UaNodeId.from(DataTypeIds.UriString), "UriString", new UaLocalizedText("UriString"), false);

    static readonly ByteString = new UaDataType(UaNodeId.from(DataTypeIds.ByteString), "ByteString", new UaLocalizedText("ByteString"), false);
    static readonly Image = new UaDataType(UaNodeId.from(DataTypeIds.Image), "Image", new UaLocalizedText("Image"), true);
    static readonly ImageBMP = new UaDataType(UaNodeId.from(DataTypeIds.ImageBMP), "ImageBMP", new UaLocalizedText("ImageBMP"), false);
    static readonly ImageGIF = new UaDataType(UaNodeId.from(DataTypeIds.ImageGIF), "ImageGIF", new UaLocalizedText("ImageGIF"), false);
    static readonly ImageJPG = new UaDataType(UaNodeId.from(DataTypeIds.ImageJPG), "ImageJPG", new UaLocalizedText("ImageJPG"), false);
    static readonly ImagePNG = new UaDataType(UaNodeId.from(DataTypeIds.ImagePNG), "ImagePNG", new UaLocalizedText("ImagePNG"), false);
    static readonly AudioDataType = new UaDataType(UaNodeId.from(DataTypeIds.AudioDataType), "AudioDataType", new UaLocalizedText("AudioDataType"), false);

    static readonly Enumeration = new UaDataType(UaNodeId.from(DataTypeIds.Enumeration), "Enumeration", new UaLocalizedText("Enumeration"), true);

    static readonly DateTime = new UaDataType(UaNodeId.from(DataTypeIds.DateTime), "DateTime", new UaLocalizedText("DateTime"), false);
    static readonly UtcTime = new UaDataType(UaNodeId.from(DataTypeIds.UtcTime), "UtcTime", new UaLocalizedText("UtcTime"), false);

    static readonly NodeId = new UaDataType(UaNodeId.from(DataTypeIds.NodeId), "NodeId", new UaLocalizedText("NodeId"), false);
    static readonly ExpandedNodeId = new UaDataType(UaNodeId.from(DataTypeIds.ExpandedNodeId), "ExpandedNodeId", new UaLocalizedText("ExpandedNodeId"), false);
    static readonly Guid = new UaDataType(UaNodeId.from(DataTypeIds.Guid), "Guid", new UaLocalizedText("Guid"), false);
    static readonly LocalizedText = new UaDataType(UaNodeId.from(DataTypeIds.LocalizedText), "LocalizedText", new UaLocalizedText("LocalizedText"), false);
    static readonly QualifiedName = new UaDataType(UaNodeId.from(DataTypeIds.QualifiedName), "QualifiedName", new UaLocalizedText("QualifiedName"), false);
    static readonly StatusCode = new UaDataType(UaNodeId.from(DataTypeIds.StatusCode), "StatusCode", new UaLocalizedText("StatusCode"), false);

    static readonly Structure = new UaDataType(UaNodeId.from(DataTypeIds.Structure), "Structure", new UaLocalizedText("Structure"), true);
    static readonly Range = new UaDataType(UaNodeId.from(DataTypeIds.Range), "Range", new UaLocalizedText("Range"), false);
    static readonly EUInformation = new UaDataType(UaNodeId.from(DataTypeIds.EUInformation), "EUInformation", new UaLocalizedText("EUInformation"), false);
    static readonly EnumValueType = new UaDataType(UaNodeId.from(DataTypeIds.EnumValueType), "EnumValueType", new UaLocalizedText("EnumValueType"), false);
}